feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/feature_layers/DisplaySubtypeFeatureLayerController.java

```diff
@@ -0,0 +1,143 @@
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+import com.esri.arcgisruntime.symbology.Symbol;
+import javafx.fxml.FXML;
+
+import com.esri.arcgisruntime.mapping.view.MapView;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ToggleButton;
+
+public class DisplaySubtypeFeatureLayerController {
+
+  @FXML private ToggleButton toggleRendererButton;
+  @FXML private MapView mapView;
+  @FXML private Label currentMapScaleLabel;
+  @FXML private Label minScaleLabel;
+  @FXML private CheckBox checkBox;
+  private SubtypeSublayer sublayer;
+  private Renderer originalRenderer;
+  
+  public void initialize() {
+
+    try {
+
+      // create a map with streets night vector basemap and add it to the map view
+      ArcGISMap map = new ArcGISMap();
+      map.setBasemap(Basemap.createStreetsNightVector());
+      mapView.setMap(map);
+      // display the current map scale 
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent ->
+        currentMapScaleLabel.setText("Current Map Scale: 1:" + Math.round(mapView.getMapScale())));
+
+      // set the viewpoint to Naperville, Illinois
+      Viewpoint initialViewpoint = new Viewpoint(new Envelope(-9812691.11079696, 5128687.20710657,
+        -9812377.9447607, 5128865.36767282, SpatialReferences.getWebMercator()));
+      map.setInitialViewpoint(initialViewpoint);
+
+      // create a subtype feature layer from the service feature table, and add it to the map
+      final String serviceFeatureTableUrl = "https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100";
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable(serviceFeatureTableUrl);
+      SubtypeFeatureLayer subtypeFeatureLayer = new SubtypeFeatureLayer(serviceFeatureTable);
+      map.getOperationalLayers().add(subtypeFeatureLayer);
+      
+      // access the json required for sublayer label definitions
+      File jsonFile = new File(System.getProperty("data.dir"),
+          Objects.requireNonNull(getClass().getClassLoader().getResource("label_definition.json")).getFile());
+      final String json;
+      // read in the complete file as a string
+      try (Scanner scanner = new Scanner(jsonFile)) {
+        json = scanner.useDelimiter("\\A").next();
+      }
+
+      subtypeFeatureLayer.loadAsync();
+      subtypeFeatureLayer.addDoneLoadingListener(() -> {
+        
+        // get the Street Light sublayer and define its labels
+        sublayer = subtypeFeatureLayer.getSublayerWithSubtypeName("Street Light");
+        sublayer.setLabelsEnabled(true);
+        sublayer.getLabelDefinitions().add(LabelDefinition.fromJson(json));
+
+        // get the original renderer of the sublayer
+        originalRenderer = sublayer.getRenderer();
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
```

Comment:
I think we've moved to not just printing the stack trace, but showing an alert with a more meaningful error message. Something like:
`new Alert(Alert.AlertType.ERROR, "Error loading the Subtype Feature Layer").show`

Reply:
in this case, the try and catch are covering the whole of the initialize method. If anything goes wrong it could be for a number of reasons, not only related to the loading of the subtype feature layer. So I think in this case it's ok to leave as `e.printStackTrace()` since we can't be certain it would only ever be a loading error.